dimod/core/structured.py

```diff
@@ -144,3 +145,19 @@ def to_networkx_graph(self):
         G.add_nodes_from(self.nodelist)
 
         return G
+
+    def valid_bqm_structure(self, bqm: "BinaryQuadraticModel"):
+        """Validate that problem defined by :class:`dimod.BinaryQuadraticModel`
+        matches the graph provided by the sampler.
+
+        Args:
+            bqm: :class:`dimod.BinaryQuadraticModel` object to validate.
+
+        Returns:
+            Boolean indicating validity of BQM structure compared to sampler graph.
+
+        """
+        if all(variable in self.adjacency for variable in bqm.variables) and \
+           all(neighbor in self.adjacency[vertex] for neighbor, vertex in bqm.quadratic):
+            return True
+        return False
```

Comment:
or even just `return all(....` :smile:

Reply:
Good point, I actually had it like that prior to pushing 🤕